File: Day20/Backend/stockx/src/main/java/com/guruprasath/stockx/service/DashboardService.java
package com.guruprasath.stockx.service;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.guruprasath.stockx.model.Dashboard;
import com.guruprasath.stockx.repository.*;

import java.util.List;
import java.util.Optional;

@Service
public class DashboardService {

    private final DashboardRepository dashboardRepository;

    @Autowired
    public DashboardService(DashboardRepository dashboardRepository) {
        this.dashboardRepository = dashboardRepository;
    }

    public List<Dashboard> getAllProducts() {
        return dashboardRepository.findAll();
    }

    public Dashboard getProductById(Long id) {
        Optional<Dashboard> productOptional = dashboardRepository.findById(id);
        return productOptional.orElse(null);
    }

    public Dashboard createProduct(Dashboard product) {
        return dashboardRepository.save(product);
    }

    public Dashboard updateProduct(Long id, Dashboard updatedProduct) {
        Optional<Dashboard> productOptional = dashboardRepository.findById(id);
        if (productOptional.isPresent()) {
            updatedProduct.setId(id);
            return dashboardRepository.save(updatedProduct);
        }
        return null; // Product not found
    }

    public void deleteProduct(Long id) {
        dashboardRepository.deleteById(id);
    }
}

File: Day22/Frontend/src/services/dashboard.service.js
import axios from "axios";

const API_URL = "http://localhost:8060/api/dashboard/";

const addProduct = (product) => {
  return axios.post(API_URL + "add", product);
};

const getAllProducts = () => {
  return axios.get(API_URL + "all");
};

const getProductById = (id) => {
  return axios.get(API_URL + id);
};



const DashboardService = {
  addProduct,
  getAllProducts,
  getProductById,

};

export default DashboardService;
